plugins/query-insights/src/main/java/org/opensearch/plugin/insights/settings/QueryInsightsSettings.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.settings;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.plugin.insights.core.exporter.QueryInsightsExporterType;
+
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Settings for Query Insights Plugin
+ *
+ * @opensearch.api
+ * @opensearch.experimental
+ */
+public class QueryInsightsSettings {
+    /**
+     * Default Values and Settings
+     */
+    public static final TimeValue MAX_WINDOW_SIZE = new TimeValue(21600, TimeUnit.SECONDS);
+    public static final int MAX_N_SIZE = 100;
+    public static final TimeValue MIN_EXPORT_INTERVAL = new TimeValue(21600, TimeUnit.SECONDS);
+    public static final String DEFAULT_LOCAL_INDEX_MAPPING = "mappings/top_n_queries_record.json";
+    /** Default window size in seconds to keep the top N queries with latency data in query insight store */
+    public static final int DEFAULT_WINDOW_SIZE = 60;
+    /** Default top N size to keep the data in query insight store */
+    public static final int DEFAULT_TOP_N_SIZE = 3;
+    /**
+     * Query Insights base uri
+     */
+    public static final String PLUGINS_BASE_URI = "/_insights";
```

Comment:
Should the base uri be in sync with plugin name and called query insights?

Reply:
Better to keep this succinct for the user? What we are gaining insights on can be inferred from the rest of the URI. Can add this point to discuss with PM during the API review

Reply:
Yeah, let us discuss this with SME since I was thinking cluster insights in mind.

Reply:
Yeah I checked with PM offline, he is okay with the namings we are introducing.